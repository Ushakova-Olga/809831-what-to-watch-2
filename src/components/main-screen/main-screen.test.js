import React from "react";
import renderer from 'react-test-renderer';
import MainScreen from '../main-screen/main-screen';

it(`Main screen correctly renders`, () => {

  const tree = renderer
    .create(<MainScreen
      films={[{
        name: `Fantastic Beasts: The Crimes of Grindelwald`,
        img: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
      },
      {
        name: `Fantastic Beasts`,
        img: `bohemian-rhapsody.jpg`,
      },
      {
        name: `Macbeth`,
        img: `macbeth.jpg`,
      },
      {
        name: `Aviator`,
        img: `aviator.jpg`,
      },
      {
        name: `We need to talk about Kevin`,
        img: `we-need-to-talk-about-kevin.jpg`,
      }
      ]} />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
