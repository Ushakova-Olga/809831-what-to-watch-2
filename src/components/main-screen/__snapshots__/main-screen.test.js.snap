// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main screen correctly renders 1`] = `
Array [
  "",
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Drama
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Comedy
          </a>
        </li>
        <li
          className="catalog__genres-item"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            Tragedy
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/0"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="100%"
                muted={true}
                poster="fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="100%"
              >
                <source
                  src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
                  type="video/mp4"
                />
              </video>
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/0"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/1"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="100%"
                muted={true}
                poster="bohemian-rhapsody.jpg"
                width="100%"
              >
                <source
                  src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
                  type="video/webm"
                />
              </video>
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/1"
              onClick={[Function]}
            >
              Fantastic Beasts
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/2"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="100%"
                muted={true}
                poster="macbeth.jpg"
                width="100%"
              >
                <source
                  src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
                  type="video/mp4"
                />
              </video>
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/2"
              onClick={[Function]}
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/3"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="100%"
                muted={true}
                poster="aviator.jpg"
                width="100%"
              >
                <source
                  src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
                  type="video/webm"
                />
              </video>
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/3"
              onClick={[Function]}
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/4"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="100%"
                muted={true}
                poster="we-need-to-talk-about-kevin.jpg"
                width="100%"
              >
                <source
                  src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
                  type="video/mp4"
                />
              </video>
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/4"
              onClick={[Function]}
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/5"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="100%"
                muted={true}
                poster="the-grand-budapest-hotel-poster.jpg"
                width="100%"
              >
                <source
                  src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
                  type="video/webm"
                />
              </video>
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/5"
              onClick={[Function]}
            >
              The Grand Budapest Hotel
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/6"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="100%"
                muted={true}
                poster="fantastic-beasts-the-crimes-of-grindelwald.jpg"
                width="100%"
              >
                <source
                  src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
                  type="video/mp4"
                />
              </video>
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/6"
              onClick={[Function]}
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            href="/films/7"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
            >
              <video
                height="100%"
                muted={true}
                poster="bohemian-rhapsody.jpg"
                width="100%"
              >
                <source
                  src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
                  type="video/webm"
                />
              </video>
            </div>
          </a>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/7"
              onClick={[Function]}
            >
              Fantastic Beasts
            </a>
          </h3>
        </article>
      </div>
      <div
        className="catalog__more"
      >
        <button
          className="catalog__button"
          onClick={[Function]}
          type="button"
        >
          Show more
        </button>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
